modules/gdscript/gdscript_parser.h

```diff
@@ -1476,8 +1477,10 @@ class GDScriptParser {
 	// Setting p_force to false will prevent the completion context from being update if a context was already set before.
 	// This should only be done when we push context before we consumed any tokens for the corresponding structure.
 	// See parse_precedence for an example.
-	void make_completion_context(CompletionType p_type, Node *p_node, int p_argument = -1, bool p_force = true);
-	void make_completion_context(CompletionType p_type, Variant::Type p_builtin_type, bool p_force = true);
+	bool make_completion_context(CompletionType p_type, Node *p_node, bool p_force = true);
+	void make_completion_context_argument(CompletionType p_type, Node *p_node, int p_argument, bool p_force = true);
+	void make_completion_context_bultin_type(CompletionType p_type, Variant::Type p_builtin_type, bool p_force = true);
+	void make_completion_context_virtual_method(CompletionType p_type, Node *p_node, bool p_has_param_parenthesis, bool p_force = true);
```

Comment:
Adding another param to `make_completion_context` with my bool felt silly and I couldn't really make an overload without colliding `p_builtin_type` with `p_has_param_parenthesis`, so I figured it's best to split those methods to keep the param count low. This also deduplicates the code a bit.

Ultimately my variant of the method didn't proved to be *that* critical, but I'd argue that this refactor makes the intent more apparent across the codebase.